cpp/src/arrow/flight/test_util.cc

```diff
@@ -474,12 +474,31 @@ class FlightTestServer : public FlightServerBase {
     return Status::OK();
   }
 
+  Status ListIncomingHeaders(const ServerCallContext& context, const Action& action,
+                             std::unique_ptr<ResultStream>* out) {
+    std::vector<Result> results;
+    auto prefix = static_cast<std::string_view>(*(action.body));
```

Comment:
nit, but IMO static_cast is a bit of a confusing way to write an implicit conversion

Reply:
OK. How about `std::string_view prefix(*action.body)`?